modules/openthread/CMakeLists.txt

```diff
@@ -433,9 +433,11 @@ target_compile_options(ot-config INTERFACE
 # errno_private.h is generated as part of ${SYSCALL_LIST_H_TARGET} target.
 add_dependencies(ot-config ${SYSCALL_LIST_H_TARGET})
 
-# Make sure C library is linked after OpenThread libraries (to prevent linker
-# errors)
-target_link_libraries(ot-config INTERFACE -lc)
+# Make sure C library, in case of newlib, is linked after OpenThread libraries
+# (to prevent linker errors)
+if(CONFIG_NEWLIB_LIBC)
+  target_link_libraries(ot-config INTERFACE -lc)
```

Comment:
My question is: why was this necessary in the first place? https://github.com/zephyrproject-rtos/zephyr/pull/48646#issuecomment-1204271028 describes what problem this is trying to fix, but does not explain the root cause.

Also is this still necessary?

Reply:
It is still required in case of `newlib`, I didn't dig deeper, just verified with `minimal/picolibc/newlib`.

Reply:
> It is still required in case of `newlib`

Ok, it would be nice if we can eventually get rid of this hack, but that is beyond the scope of this PR.